resources/dw/model_cluster.go

```diff
@@ -0,0 +1,136 @@
+	InstanceSettings            *azureInstanceResourceModel       `tfsdk:"instance_settings"`
+	AdvancedSettings            *clusterAdvancedResourceModel     `tfsdk:"advanced_settings"`
+}
+
+func (p *awsClusterResourceModel) convertToCreateAwsClusterRequest() *models.CreateAwsClusterRequest {
+	awsClusterRequest := &models.CreateAwsClusterRequest{
+		EnvironmentCrn:                   p.Crn.ValueStringPointer(),
+		UseOverlayNetwork:                p.NetworkSettings.GeneralSettings.UseOverlayNetwork.ValueBool(),
+		WhitelistK8sClusterAccessIPCIDRs: utils.FromListValueToStringList(p.NetworkSettings.GeneralSettings.WhitelistK8sClusterAccessIpCidrs),
+		WhitelistWorkloadAccessIPCIDRs:   utils.FromListValueToStringList(p.NetworkSettings.GeneralSettings.WhitelistWorkloadAccessIpCidrs),
+		UsePrivateLoadBalancer:           p.NetworkSettings.GeneralSettings.UsePrivateLoadBalancer.ValueBool(),
+		UsePublicWorkerNode:              p.NetworkSettings.GeneralSettings.UsePublicWorkerNode.ValueBool(),
+		WorkerSubnetIds:                  utils.FromListValueToStringList(p.NetworkSettings.WorkerSubnetIds),
+		LbSubnetIds:                      utils.FromListValueToStringList(p.NetworkSettings.LbSubnetIds),
+		NodeRoleCDWManagedPolicyArn:      p.NodeRoleCDWManagedPolicyArn.ValueString(),
+		ReducedPermissionMode:            p.ReducedPermissionMode.ValueBool(),
+	}
+
+	if p.AdvancedSettings != nil {
+		awsClusterRequest.DatabaseBackupRetentionPeriod = utils.Int64To32Pointer(p.AdvancedSettings.DatabaseBackupRetentionPeriod)
+		awsClusterRequest.ReservedComputeNodes = utils.Int64To32(p.AdvancedSettings.ReservedComputeNodes)
+		awsClusterRequest.ReservedSharedServicesNodes = utils.Int64To32(p.AdvancedSettings.ReservedSharedServicesNodes)
+		awsClusterRequest.CustomSubdomain = p.AdvancedSettings.CustomSubdomain.ValueString()
+		if p.AdvancedSettings.CustomRegistryOptions != nil {
+			awsClusterRequest.CustomRegistryOptions = &models.CustomRegistryOptions{
+				DisableImageVerification: p.AdvancedSettings.CustomRegistryOptions.DisableImageVerification.ValueBool(),
+				Password:                 p.AdvancedSettings.CustomRegistryOptions.Password.ValueString(),
+				UserName:                 p.AdvancedSettings.CustomRegistryOptions.Username.ValueString(),
+				RegistryType:             p.AdvancedSettings.CustomRegistryOptions.RegistryType.ValueString(),
+				RepositoryURL:            p.AdvancedSettings.CustomRegistryOptions.RepositoryURL.ValueString(),
+			}
+		}
+	}
+
+	if p.InstanceSettings != nil {
+		awsClusterRequest.EnableSpotInstances = p.InstanceSettings.EnableSpotInstances.ValueBoolPointer()
+		awsClusterRequest.CustomAmiID = p.InstanceSettings.CustomAmiId.ValueString()
+		awsClusterRequest.ComputeInstanceTypes = utils.FromListValueToStringList(p.InstanceSettings.ComputeInstanceTypes)
+		awsClusterRequest.AdditionalInstanceTypes = utils.FromListValueToStringList(p.InstanceSettings.AdditionalInstanceTypes)
+	}
+
+	return awsClusterRequest
```

Comment:
when nested structures have been extracted to a function the end of the road it'll look like
```suggestion
	return &models.CreateAwsClusterRequest{
```